modules/prometheus-source/pkg/prom/datasource.go

```diff
@@ -132,11 +133,22 @@ func NewPrometheusDataSource(infoProvider clusters.ClusterInfoProvider, promConf
 
 	// we don't consider this a fatal error, but we log for visibility
 	api := prometheusAPI.NewAPI(promClient)
-	_, err = api.Buildinfo(context.Background())
+	bi, err := api.Buildinfo(context.Background())
+
 	if err != nil {
 		log.Infof("No valid prometheus config file at %s. Error: %s.\nTroubleshooting help available at: %s.\n**Ignore if using cortex/mimir/thanos here**", promConfig.ServerEndpoint, err.Error(), PrometheusTroubleshootingURL)
 	} else {
 		log.Infof("Retrieved a prometheus config file from: %s", promConfig.ServerEndpoint)
+		promConfig.Version = bi.Version
+
+		// for versions of prometheus >= 3.0.0, we need to offset the resolution for range queries
+		// due to a breaking change in prometheus lookback and range query alignment
+		v, err := semver.NewVersion(promConfig.Version)
```

Comment:
If this is the only place we're using the semver pkg, would it be worthwhile to just write/copy the helper function ourselves as opposed to importing an additional pkg?

If not, should we at least go up to `semver/v3`? It seems v1 is no longer supported ([ref](https://github.com/Masterminds/semver?tab=readme-ov-file#package-versions))

Reply:
> If this is the only place we're using the semver pkg, would it be worthwhile to just write/copy the helper function ourselves as opposed to importing an additional pkg?

until they add a `v` in front or some other garbage that breaks all the logic? No thanks. Using semver will at least guarantee that as long as they're returning a valid semver version, we can parse it and pull out the major version.

Reply:
Haha, good point!